plumbing/object/commit_walker.go

```diff
@@ -184,6 +184,76 @@ func (w *commitPostIterator) ForEach(cb func(*Commit) error) error {
 
 func (w *commitPostIterator) Close() {}
 
+type commitPostIteratorFirstParent struct {
+	stack []*Commit
+	seen  map[plumbing.Hash]bool
+}
+
+// NewCommitPostorderIter returns a CommitIter that walks the commit
+// history like WalkCommitHistory but in post-order. This means that after
+// walking a merge commit, the merged commit will be walked before the base
+// it was merged on. This can be useful if you wish to see the history in
+// chronological order. Ignore allows to skip some commits from being iterated.
```

Comment:
```suggestion
// NewCommitPostorderIterFirstParent returns a CommitIter that walks the commit
// history like WalkCommitHistory but in post-order. 
// 
// This option gives a better overview when viewing the evolution of a particular
// topic branch, because merges into a topic branch tend to be only about
// adjusting to updated upstream from time to time, and this option allows
// you to ignore the individual commits brought in to your history by such
// a merge.
// 
// Ignore allows to skip some commits from being iterated.
```

Reply:
It feels focusing specifically on back-merges from main branches into topic branches may be too specific - This change basically returns only commits that are directly on the branch that is being traversed. This essentially means that, regardless of the selected branch (be that main or a feature branch), when a merge commit is encountered, only the merge commit itself and the base commit (i.e. the last shared commit before the branches diverged from main) will be traversed.